src/sql/src/plan/query.rs

```diff
@@ -3521,18 +3654,41 @@ fn plan_identifier(ecx: &ExprContext, names: &[Ident]) -> Result<HirScalarExpr,
         // The name refers to a normal table. Return a record containing all the
         // columns of the table.
         _ => {
+            let mut has_exists_column = None;
             let (exprs, field_names): (Vec<_>, Vec<_>) = items
                 .into_iter()
-                .map(|(column, item)| {
-                    let expr = HirScalarExpr::Column(column);
-                    let name = item.column_name.clone();
-                    (expr, name)
+                .filter_map(|(column, item)| {
+                    if item.is_exists_column_for_a_table_function_that_was_in_the_target_list {
+                        has_exists_column = Some(column);
+                        None
+                    } else {
+                        let expr = HirScalarExpr::Column(column);
+                        let name = item.column_name.clone();
+                        Some((expr, name))
+                    }
                 })
                 .unzip();
-            Ok(HirScalarExpr::CallVariadic {
-                func: VariadicFunc::RecordCreate { field_names },
-                exprs,
-            })
+            let expr = if exprs.len() == 1 && has_exists_column.is_some() {
+                // TODO: this probably isn't correct, do the same thing resolve_columns does maybe? Dunno.
+                exprs[0].clone()
```

Comment:
Is this right for multi-column-returning functions? Or is this a code path where those already been converted to records at this point?

Reply:
The mult-column functions are handled in the else branch here.